packages/core/src/__storage__/base.ts

```diff
@@ -0,0 +1,22 @@
+import { WebStorageManager } from "./web/storage";
+import { isReactNative, isWeb } from "@utils";
+import type { StorageBase } from "@types";
+
+// TODO (Amir): Turn this into a class that extends StorageBase and make an init function. See stamper
+export async function createStorageManager(): Promise<StorageBase> {
+  if (isReactNative()) {
+    try {
+      // Dynamic import to prevent bundling the native module in web environments
+      const { MobileStorageManager } = await import("./mobile/storage");
+      return new MobileStorageManager();
+    } catch (error) {
+      throw new Error(
+        `Failed to load storage manager for react-native: ${error}`,
+      );
+    }
+  } else if (isWeb()) {
+    return new WebStorageManager();
+  } else {
+    throw new Error("Unsupported environment for storage manager.");
+  }
+}
```

Comment:
It'd be nice to avoid making assumptions when it comes to what users will / will not import. Do you think it's possible to let users import their own storage manager as they need it instead of have a createStorageManager which just wraps the thing that they'll need?

The value we're providing with this thin layer is very low (and can be frustrating for users if we get the `isWeb` or `isReactNative` checks wrong)